### PR TITLE
Feature/bare bones

### PR DESCRIPTION
**Pull request by nick-shaw**
Strip down to 'bare bones' version, with power(p) curve only, and with shadow roll-off and 'hexagonal' mode removed.

### CODE REVIEW

File: model/GamutCompress.blink [hunk 1]
```diff
@@ -28,163 +25,28 @@ kernel GamutCompression : public ImageComputationKernel<ePixelWise> {
```
Comment (thomasmansencal) [inline model/GamutCompress.blink:22]: Github is lame here, but we could certainly trim `float pi;` and `pi = 3.14159265359;` as they are not used anymore.

File: model/GamutCompress.glsl [hunk 2]
```diff
@@ -1,7 +1,7 @@
 uniform sampler2D frontTex, matteTex, selectiveTex;
-uniform float power, cyan, magenta, yellow, shd_rolloff, adsk_result_w, adsk_result_h;
-uniform int method, working_colorspace;
-uniform bool invert, hexagonal, overlay;
+uniform float power, cyan, magenta, yellow, adsk_result_w, adsk_result_h;
+uniform int working_colorspace;
+uniform bool invert, overlay;
 uniform vec3 threshold;
 
 const float pi = 3.14159265359;
```
Comment (thomasmansencal) [inline model/GamutCompress.glsl]: Same here, can go.

Comment (thomasmansencal) [inline model/GamutCompress.glsl]: Can go too along with `tanh` and `atanh`.

**Comment by thomasmansencal** [comment]
Haven't tested but overall LGTM, there are some trigonometric function and requirements left over here and there, e.g. PI definitions in a lot of files.

Quite like how it is leaner now! 👍

**Review by carolalynn — APPROVED** [review]
Tested in Nuke and Resolve, all seems well. Thanks Nick :)

**Comment by nick-shaw** [comment]
I will clean up more tomorrow before merging

File: model/GamutCompress.nk [hunk 3]
```diff
@@ -96,68 +83,31 @@ set N59f7d000 [stack 0]
...
   addUserKnob {7 cyan}
-  cyan 0.09
+  cyan {{parent.compress.cyan}}
   addUserKnob {7 magenta}
-  magenta 0.24
+  magenta {{parent.compress.magenta}}
   addUserKnob {7 yellow}
-  yellow 0.12
+  yellow {{parent.compress.yellow}}
   addUserKnob {26 ""}
```
Comment (jedypod) [inline model/GamutCompress.nk:110]: Now that we've removed the bisection solver, the `lim_x, lim_y, lim_z` knobs don't do anything anymore and could be removed.

File: model/GamutCompress.nk [hunk 4]
```diff
@@ -485,56 +195,45 @@ push $N59fe9800
...
-  cyan {{parent.cyan}}
+  cyan 0.09
   addUserKnob {7 magenta}
-  magenta {{parent.magenta}}
+  magenta 0.24
   addUserKnob {7 yellow}
-  yellow {{parent.yellow}}
+  yellow 0.12
   addUserKnob {26 ""}
   addUserKnob {7 lim_x R -5 2}
```
Comment (jedypod) [inline model/GamutCompress.nk]: Here as well these lim_* knobs could be removed.

File: model/GamutCompress.nk [hunk 5]
```diff
@@ -915,12 +343,11 @@ push $N1c3d9b80
   xpos -40
   ypos 230
  }
-set N5a1f5800 [stack 0]
+set N702b7000 [stack 0]
  Merge2 {
   inputs 2
   operation screen
```
Comment (jedypod) [inline model/GamutCompress.nk]: One other small thing: Using a screen here is causing negative values in the image to be clamped to 0.0 when plot overlays is enabled.

Using a Plus or an Over instead would fix the issue.

**Comment by thomasmansencal** [comment]
LGTM! Looks like we are done @nick-shaw right?

**Comment by nick-shaw** [comment]
I will change the default values to those from the Python notebook, and change the link in the ReadMe to point to Matthias’ version of it. Then we can merge.

**Pull request merged**